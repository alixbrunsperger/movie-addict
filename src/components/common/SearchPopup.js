import React from 'react';

export default ({isPopupActive, hidePopup}) => (
      <div className={`search-popup ${isPopupActive ? "active" : ""}`}>
        <div className="search-block">
          <img src="/images/cross.png" onClick={hidePopup} alt="close icon" />
          <form action="/results">
            <input type="text" name="query" />
            <input type="submit" value="Search" />
          </form>
        </div>
      </div>
  );
  
