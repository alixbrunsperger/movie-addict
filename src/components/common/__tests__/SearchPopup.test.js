import React from 'react';
import { shallow } from 'enzyme';
import SearchPopup from '../SearchPopup';
describe('SearchPopup', () => {
  it('is rendered', () => {
    const component = shallow(<SearchPopup />);
    expect(component).toMatchSnapshot();
  });

describe('prop isPopupActive', () => {
  it('when isPopupActive is true is rendered', () => {
    const component = shallow(<SearchPopup isPopupActive />);
    expect(component).toMatchSnapshot();
  });

  it('when isPopupActive is false is rendered', () => {
    const component = shallow(<SearchPopup isPopupActive={false} />);
    expect(component).toMatchSnapshot();
  });
});

  it('pass down hidePopup prop', () => {
    const hidePopup= jest.fn();
    const component = shallow(<SearchPopup hidePopup={hidePopup}/>);
    expect(component.find("img").prop("onClick")).toEqual(hidePopup);
  });
});
