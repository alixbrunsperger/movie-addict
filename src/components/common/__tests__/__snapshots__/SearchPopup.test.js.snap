// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPopup is rendered 1`] = `
<div
  className="search-popup "
>
  <div
    className="search-block"
  >
    <img
      alt="close icon"
      src="/images/cross.png"
    />
    <form
      action="/results"
    >
      <input
        name="query"
        type="text"
      />
      <input
        type="submit"
        value="Search"
      />
    </form>
  </div>
</div>
`;

exports[`SearchPopup prop isPopupActive when isPopupActive is false is rendered 1`] = `
<div
  className="search-popup "
>
  <div
    className="search-block"
  >
    <img
      alt="close icon"
      src="/images/cross.png"
    />
    <form
      action="/results"
    >
      <input
        name="query"
        type="text"
      />
      <input
        type="submit"
        value="Search"
      />
    </form>
  </div>
</div>
`;

exports[`SearchPopup prop isPopupActive when isPopupActive is true is rendered 1`] = `
<div
  className="search-popup active"
>
  <div
    className="search-block"
  >
    <img
      alt="close icon"
      src="/images/cross.png"
    />
    <form
      action="/results"
    >
      <input
        name="query"
        type="text"
      />
      <input
        type="submit"
        value="Search"
      />
    </form>
  </div>
</div>
`;
